presentation-compiler/src/main/dotty/tools/pc/printer/ShortenedTypePrinter.scala

```diff
@@ -166,6 +165,37 @@ class ShortenedTypePrinter(
           res.toPrefixText
       }
 
+  /** Find a shortened prefix for a symbol which can be accessed statically in current run.
+   *  @param sym - statically accessible symbol
+   *
+   *  @return shortened prefix for sym
+   */
+  private def missingSymbolPrefixOf(sym: Symbol): Text =
+
+    def toTextPrefix0(owner: Symbol, acc: List[Symbol], wasPreviousAConflict: Boolean): List[Symbol] =
+      if !owner.exists then acc
+      else if owner.is(Synthetic) then toTextPrefix0(owner.maybeOwner, acc, wasPreviousAConflict)
+      else
+        indexedCtx.lookupSym(owner) match
+          case Result.Conflict => toTextPrefix0(owner.maybeOwner, owner :: acc, wasPreviousAConflict = true)
+          // Previous owner was at conflict so we have to include current one in prefix to disambiguate
+          case _ if wasPreviousAConflict => toTextPrefix0(owner.maybeOwner, owner :: acc, wasPreviousAConflict = false)
+          // If symbol is defined in the current run, we want to strip the symbol until first InScope is found
+          case Result.InScope if sym.isDefinedInCurrentRun => owner :: acc
+          // We want to avoid package names in the shortened names
+          case Result.InScope if owner.is(Package) || owner.isPackageObject => acc
+          case Result.Missing if owner.is(Package) || owner.isPackageObject =>
+            // if current owner is nested, we want to import current owner and return collected prefix
+            if owner.maybeOwner.exists && acc.nonEmpty then
```

Comment:
Why does it matter if `owner` has an owner?

Reply:
We're at the root, and it is better to import symbol instead of using fully qualified path. As I mentioned in the PR, it should be configurable, but I'm not sure if this is scope of this PR.